feat: ✨ implemented population calculation functionality

de-vasta opened the pull request.

de-vasta committed:
feat: ✨ implemented population calculation functionality

mateacademy-ai-mentor commented on src/scripts/main.js:
```diff
@@ -1,3 +1,16 @@
+const avgPopulation = totalPopulation / populations.length;
+const avgPopulationEl = document.querySelector('.average-population');
+
+avgPopulationEl.textContent = avgPopulation.toLocaleString('en-US');
```
The calculation for the average population might result in a number with a decimal part. According to the requirements, the output should have the 'same numeric style as populations presented,' which are integers. Consider rounding the `avgPopulation` value before formatting it to match the required style.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Great work on this task! The logic for calculating the total and average population is correct, and you've used modern JavaScript methods effectively.

One change is needed to meet the project's formatting requirements. The average population is currently displayed with decimal places, but it should be a whole number to match the style of the original data. 

Please update your code to round the `avgPopulation` value before converting it to a string. Using `Math.round()` on the result of your division would be a perfect solution here. After that small fix, this will be ready to go!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

de-vasta committed:
fix: 🛠️ improved population calculation accuracy and data filtering

- Added filter to remove **NaN** values from population data parsing
- Applied `Math.trunc` to average population calculation for integer result

mateacademy-ai-mentor commented on src/scripts/main.js:
```diff
@@ -1,3 +1,16 @@
+
+totalPopulationEl.textContent = totalPopulation.toLocaleString(locale);
+
+const avgPopulation = Math.trunc(totalPopulation / populations.length);
```
Using `Math.trunc()` successfully makes the average a whole number, which is great progress! However, `Math.round()` is generally a better choice for rounding an average, as it rounds to the nearest whole number. `Math.trunc()` always rounds towards zero (or down for positive numbers), which can be slightly less accurate.

Please use `Math.round()` here as suggested in the previous review.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Excellent work on this task! I am approving your solution as it successfully meets all the core requirements. The logic for calculating and displaying the population data is implemented correctly.

As a final suggestion, while `Math.trunc()` does produce a whole number, it's standard practice to use `Math.round()` for averages. This rounds to the nearest integer, providing a more representative value. Your current code is fully functional, so this is just a tip for future reference.

Great job!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

de-vasta committed:
Replaced `Math.trunc()` with `Math.round()` for more accurate average population calculation
• This ensures the average is properly rounded to the nearest integer instead of being truncated
